homeassistant/components/alarm_control_panel/simplisafe.py

```diff
@@ -37,36 +36,23 @@
 
 def setup_platform(hass, config, add_devices, discovery_info=None):
     """Set up the SimpliSafe platform."""
-    from simplipy.api import SimpliSafeApiInterface, get_systems
+    from simplipy.api import SimpliSafeApiInterface, SimpliSafeAPIException
     name = config.get(CONF_NAME)
     code = config.get(CONF_CODE)
     username = config.get(CONF_USERNAME)
     password = config.get(CONF_PASSWORD)
 
-    simplisafe = SimpliSafeApiInterface()
-    status = simplisafe.set_credentials(username, password)
-    if status:
-        hass.data[DOMAIN] = simplisafe
-        locations = get_systems(simplisafe)
-        for location in locations:
-            add_devices([SimpliSafeAlarm(location, name, code)])
-    else:
-        message = 'Failed to log into SimpliSafe. Check credentials.'
-        _LOGGER.error(message)
-        hass.components.persistent_notification.create(
-            message,
-            title=NOTIFICATION_TITLE,
-            notification_id=NOTIFICATION_ID)
-        return False
-
-    def logout(event):
-        """Logout of the SimpliSafe API."""
-        hass.data[DOMAIN].logout()
-
-    hass.bus.listen(EVENT_HOMEASSISTANT_STOP, logout)
-
-
-class SimpliSafeAlarm(alarm.AlarmControlPanel):
+    try:
+        simplisafe = SimpliSafeApiInterface(username, password)
+    except SimpliSafeAPIException:
+        _LOGGER.error("Failed to setup SimpliSafe")
+        return
+
+    for system in simplisafe.get_systems():
+        add_devices([SimpliSafeAlarm(system, name, code)])
```

Comment:
Add all devices in one call to `add_devices`. Sorry I missed this before.

Reply:
No problem. I assume this is an efficiency thing? I always forget to do this.